language/libra-vm/src/transaction_metadata.rs

```diff
@@ -79,6 +86,7 @@ impl Default for TransactionMetadata {
             gas_unit_price: GasPrice::new(0),
             transaction_size: AbstractMemorySize::new(0),
             expiration_time: Duration::new(0, 0),
+            chain_id: ChainId::test(),
```

Comment:
I think this may cause issues on mainnet without an update to `process_block_prologue` in LibraVM. In particular, it currently uses `TransactionMetatdata::default` when spinning up a block prologue. So we need to make sure that the chain_id field is set correctly there. I think it's going to need to read the `ChainId` resource from chain for that though. @runtian-zhou @dariorussi  what are your thoughts on this?

Reply:
I hate the `Default` trait...

Reply:
Took a look and it seems that the `process_block_prologue` code path doesn't read the `chain_id` from `txn_data` and doesn't need to. If BlockMetadata were a user-submitted transaction, we would definitely need to check the chain ID there to prevent replays. But since BlockMetadata txes are only created/executed inside the VM, I think we are safe from replays.

Reply:
It looks like `process_block_prologue` actually shouldn't need `txn_data` at all; perhaps Runtian's refactor will fix that.

Reply:
ah! I see. Totally makes sense, and this doesn't go through prologue etc. Good call.